src/roswire/app/instance.py

```diff
@@ -161,6 +164,16 @@ def ros2(self) -> ROS2:
         """Provides access to ROS2 inside this application instance."""
         return ROS2.for_app_instance(self)
 
+    @property
```

Comment:
ROS1 + ROS2

Reply:
I don't really think this belongs here since `AppInstance` isn't related to launch activities (i.e., we're reducing coherence).

Reply:
I have refactored this into roswire.proxy.roslaunch.factories. I think #407 will be handle things in similar ways.